src/python/pants/vcs/git_test.py

```diff
@@ -602,5 +602,5 @@ async def worktree_id_string() -> str:
 )
 def test_parse_unified_diff(diff, expected):
```

Comment:
It'd be good to expand this test to validate the new behaviour / stop regressions, by adding a new case with an example of a problematic diff. Let me know if I can help you make this happen! 😄